docs/style/scala.md

```diff
@@ -0,0 +1,106 @@
+Raster Foundry Style Guide
+============
+
+Comments
+--------
+
+Single line comments should be wrapped in `/**...*/`.
+
+```scala
+/** good */
+// bad
+```
+
+Multi-line comments should start with a `/**`, have following lines asterisks trail
+the _second_ asterisk from the first line, and end with a `*/`:
+
+```scala
+/** This is a good multi-line comment
+  *
+  * good stuff
+  */
+```
+
+```scala
+/** This is a bad multi-line comment
+ *
+ *  bad stuff
+ */
+```
+
+This is _different_ from Scaladoc's style, but it's how we've done things, and it's
+more consistent with autoindentation in IntelliJ's IDEA and the spacemacs scala
+layer.
+
+Class Declarations
+---------
+
+When a class extends many traits, the subsequent `with foo` blocks should be
+indented four lines, with the last one followed by the opening curly brace.
+
+```scala
+class GoodClass extends Foo
+    with Bar
+    with Baz
+    with Qux {
+    ...
+}
+```
+
+"Many traits" means the class declaration would be over 100 characters, though if
+it's getting close and readability would be improved you should go ahead and separate
+the lines anyway.
+
+Long-hanging indents before `with` statements can be replaced with
+`sed "s/\\s\{n\}\(with.*\)/    \1/" filename > filename.styled`,
+optionally `sed -i` if you're feeling lucky.
+
+Endpoints and Routes
+---------
+
+#### Methods
+
+Folowing route definition, methods should be the outermost block of code, e.g.:
+
+```scala
+get {
+  stuff {
+    ...more stuff...
+  }
+} ~
+post {
+  authenticate {
+    ...stuff...
+  }
+} ~ ...
+```
+
+Not:
+
+```scala
+authenticate {
+  post {
+    ...more stuff...
+  }
+} ~
+stuff {
+  get {
+    ...more stuff...
+  }
+} ~ ...
+```
+
+Methods as the outermost block puts allowed methods in a consistent place and also
+prevents the problem discussed in [issue #562](https://github.com/azavea/raster-foundry/issues/562/).
+
+#### Routes
+
+Endpoints for returning several objecst are `list` endpoints, and should be named `list{object}` as
+opposed to `get{object}s`, since `get{objects}s` is hard to distinguish visually from `get{object}`.
+([Issue #475](https://github.com/azavea/raster-foundry/issues/475))
+
```

Comment:
I can suggest an addition here: Push the static parts of the routes outside of the individual route objects. This gives a route object that's more representative of the overall routing structure and gets rid of some nesting in the specific route objects themselves,.

instead:
```scala
val routes = cors() {
  healthCheckRoutes ~
    userRoutes ~
    organizationRoutes ~
    sceneRoutes ~
    bucketRoutes ~
    imageRoutes ~
    tokenRoutes ~
    thumbnailRoutes ~
    configRoutes
}
```

```scala
val routes = cors () {
  pathPrefix("healthcheck") {
    healthCheckRoutes
  } ~
  pathPrefix("api") {
    pathPrefix("buckets") {
      bucketRoutes
    } ~
    pathPrefix("images") {
      imageRoutes
    }
  } ~
  ???
}
```